File: C++/ABC/127/C.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n, m, l, r;
    cin >> n >> m >> l >> r;
    for (int i = 0; i < m - 1; ++i) {
        int l_, r_;
        cin >> l_ >> r_;
        l = max(l, l_);
        r = min(r, r_);
    }
    if (r - l + 1 < 0) {
        cout << 0 << endl;
    } else {
        cout << r - l + 1 << endl;
    }
}

File: Python/ABC/127/C.py
import sys

sys.setrecursionlimit(10 ** 5)

n, m = map(int, input().split())

l, r = map(int, input().split())


def check(i, l, r):
    if i == m:
        return max(r - l + 1, 0)
    l_, r_ = map(int, input().split())
    nl = max(l, l_)
    nr = min(r, r_)
    i += 1
    return check(i, nl, nr)


print(check(1, l, r))